examples/example_ecostress.py

```diff
@@ -0,0 +1,70 @@
+#Copyright (C) 2017-2018 COAL Developers
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 2.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public
+# License along with this program; if not, write to the Free
+# Software Foundation, Inc., 51 Franklin Street, Fifth
+# Floor, Boston, MA 02110-1301, USA.
+# encoding: utf-8
+
+'''
+example_ecostress.py -- a script which will generate envi .sli and .hdr convolved library
+files of `ECOStress Spectral Library <https://speclib.jpl.nasa.gov/>`_
+
+Dependencies
+`ECOStress Spectral Library <https://speclib.jpl.nasa.gov/>`_
+must be downloaded to the examples directory
+
+All files generated will be located in the examples directory
+
+
+@author:     COAL Developers
+
+@copyright:  2018 COAL Developers. All rights reserved.
+
+@license:    GNU General Public License version 2
+
+@contact:    coal-capstone@googlegroups.com
+'''
+
+#!/usr/bin/python
+
+import os
+import sys
+import pycoal
+sys.path.insert(0, '../pycoal')
+import mineral
+import fnmatch
+
+#Path to EcoStress Spectral Library
+if (len(sys.argv) == 1):
+    raise Exception('Must pass in location of EcoStress Spectral Library')
+library_filename = sys.argv[1]
+#Loop through and edit all .spectrum.txt files
+for root, dir, files in os.walk(library_filename):
+    dir[:] = [d for d in dir]
+    for items in fnmatch.filter(files, "*spectrum.txt"):
+        #Remove vegation files because their format causes errors in the Spectral class
+        if "vegetation" in items:
```

Comment:
Can you remind me again why we are not using the vegetation indexes?

Reply:
On lines 4 & 5 Vegetation Spectra has Genus and Species instead of Subclass and Particle Size entries like other Spectra. 
The ASTER conversion function throws an error when it reads lines 4 & 5 so I just removed the vegetation files. 
If we wanted to read Vegetation files then we would need to change lines 4 & 5 to give them a subclass and particle size with a default value like I did with Spectral 7 conversion function. I know you said you prefer if we not plug in default values. Let me know if you want me to change this, it wouldn't be very hard

Reply:
manmade.concrete.constructionconcrete.solid.all.0598uuucnc.jhu.becknic.spectrum.txt

Name: Construction  Concrete
Type: manmade
Class: Concrete
Subclass: Construction Concrete
Particle Size: Solid
Sample No.: 0598UUUCNC

Reply:
nonphotosyntheticvegetation.bark.abies.concolor.all.vh311.ucsb.asdnicolet.spectrum.txt

Name: Abies concolor bark
Type: non photosynthetic vegetation
Class: bark
Genus: Abies
Species: concolor
Sample No.: VH311

Reply:
OK, ideally we want to be able to use every spectra from the entire ECOSTRESS library. If you could implement the suggested workaround for including the vegetation files it would be appreciated.

Reply:
@lewismc Ok, I fixed it so it works with Vegetation files. I added the Subclass and Particle Size and removed the Genus and Species. It can now convert the entire EcoStress library